webserver/websocket/src/main/java/io/helidon/webserver/websocket/WsConnection.java

```diff
@@ -75,6 +78,13 @@ private WsConnection(ConnectionContext ctx,
         this.listener = wsRoute.listener();
         this.dataReader = ctx.dataReader();
         this.lastRequestTimestamp = DateTime.timestamp();
+        this.wsConfig = (WsConfig) ctx.listenerContext()
+                                      .config()
+                                      .protocols()
+                                      .stream()
+                                      .filter(p -> p instanceof WsConfig)
```

Comment:
You could work around the ugly cast by doing (after filter)
`.map(WsConfig.class::cast)`
which will change the stream to correctly typed one.